src/core/ext/transport/chttp2/client/secure/secure_channel_create.c

```diff
@@ -97,15 +98,15 @@ grpc_channel *grpc_secure_channel_create(grpc_channel_credentials *creds,
   GRPC_API_TRACE(
       "grpc_secure_channel_create(creds=%p, target=%s, args=%p, "
       "reserved=%p)",
-      4, (creds, target, args, reserved));
+      4, ((void *)creds, target, (void *)args, (void *)reserved));
   GPR_ASSERT(reserved == NULL);
   // Make sure security connector does not already exist in args.
```

Comment:
David, in light of Julien's comments about all security connectors needing to know the right target name, I'd like to renew my suggestion of moving the security connector creation into client_channel_factory_create_subchannel().  While it's true that this would result in a separate security connector for each subchannel, I think that would allow us to centralize the use of the GRPC_ARG_BALANCER_NAME_MAP channel arg in one place, so that the target name lookup doesn't have to be done in each individual security connector implementation.

Julien, would there be any adverse affects of this approach, in terms of either memory usage or performance?  In particular, I'm thinking of situations where there are a large number of servers (and hence a large number of subchannels).

Other thoughts?